sources/web/datalab/polymer/modules/Utils.ts

```diff
@@ -112,4 +112,20 @@ class Utils {
     }
   }
 
+  /**
+   * Called when an API call fails, so that a connection error message can be shown.
+   */
+  static connectionFailed() {
+    const host = <DatalabAppElement>document.querySelector('datalab-app');
+    host.showNotification('Failed to connect to the server.', true /*sticky*/);
+  }
+
+  /**
+   * Called when an API call succeeds, so that the connection error message can be hidden.
+   */
+  static connectionSucceeded() {
+    const host = <DatalabAppElement>document.querySelector('datalab-app');
+    host.hideNotification();
```

Comment:
Whatever happens last wins. If the last xhr request fails then the toast appears with the error message, if the last one succeeded, this means we really are connected, so the toast should disappear.
Or am I missing something?